src/@next/Banner/Banner.tsx

```diff
@@ -58,37 +61,60 @@ export const Banner = React.forwardRef<HTMLDivElement, BannerProps>(
         ref={ref}
         data-titled={!!title}
         data-status={status}
+        data-type={type}
         {...props}
       >
-        <StyledBannerTitleContainer>
-          <Icon name={iconNameValue} />
-          {title && (
-            <StyledBannerTitle>
-              <Typography as="span" variant="subtitle1">
-                {title}
-              </Typography>
-            </StyledBannerTitle>
-          )}
-          {dismissable && (
-            <StyledCloseIconWrapper role="button" onClick={() => onDismiss?.()}>
-              <Icon name="ri-close" />
-            </StyledCloseIconWrapper>
-          )}
-        </StyledBannerTitleContainer>
-        <StyledBannerContentContainer>
+        {type === 'static' ? (
+          <StyledBannerTitleContainer>
+            <Icon name={iconNameValue} />
+            {title && (
+              <StyledBannerTitle data-type={type}>
+                <Typography as="span" variant="subtitle1">
+                  {title}
+                </Typography>
+              </StyledBannerTitle>
+            )}
+            {dismissable && (
+              <StyledCloseIconWrapper
+                role="button"
+                onClick={() => onDismiss?.()}
+              >
+                <Icon name="ri-close" />
+              </StyledCloseIconWrapper>
+            )}
+          </StyledBannerTitleContainer>
+        ) : (
+          type === 'fixed' && (
+            <StyledBannerContentContainer data-type={type}>
+              <Icon name={iconNameValue} />
+            </StyledBannerContentContainer>
+          )
+        )}
```

Comment:
Just wanted to ask for suggestions...

Do you think it is confusing to see the logic like this (and please refer to the `BannerStyle` file too)? Was contemplating creating an entirely different styled div for the fixed one, but it made a few things repeated (though can be more readable and less confusing to edit fixed or static seperately, as the static banner is quite distinct too in terms of stylings compared to the fixed one)

e.g., 
```
if (type === "static") return <>usual</>
else return <>here I create new styled div with all the fixed 
```

Reply:
Are those 2 types entirely different component or just the style is differ?

Reply:
the component are the same but the ordering (and some of the child component) could be a bit different too (which makes me think if I should seperate them) 

For example, before it was A B C D
And now I want to make it B C D A, with some extra rules (conditionals) whether it exists or not

what annoys me is that there are many conditionals there of whether it is static or fixed or fixed-noicon, I feel it made the code a bit more confusing compared to before, wdyt? is this okay?

Reply:
Ah ic, yeah better to separate them and use early return for readability

Reply:
Needed for line 87 to 91 in `Banner.tsx`

```
          type === 'fixed' && (
            <StyledBannerContentContainer data-type={type}>
              <Icon name={iconNameValue} />
            </StyledBannerContentContainer>
          )
```

Essentially, `fixed-noicon` is just `fixed` but without any icon 🤣 
It's different than not giving the `icon` prop though (I can't just check if `icon` prop is undefined or not). The default behavior is that if we don't give an icon prop, a default icon will be assigned depending on the `status` prop.

Was thinking to create another boolean prop (e.g., `hasIcon`), but since based on Figma this should only be possible for fixed type, so yeah I did this instead.